### PR TITLE
#9972 Fix cross layer filter persistence bug

### PR DESCRIPTION
**Pull request by Igi-ID**
## Description
Tweaked LOAD_FILTER reducer state assembly to always set the crossLayerExpanded flag,
either explicitly by the passed action or default.state as fallback. Fixes persistence bug.
Documented decision.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## Issue

**What is the current behavior?**
[#9972](https://github.com/geosolutions-it/MapStore2/issues/9972)

**What is the new behavior?**
The cross layer filter is persisted.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/reducers/queryform.js [hunk 1]
```diff
@@ -590,12 +590,17 @@ function queryform(state = initialState, action) {
...
             ...{
                 attributePanelExpanded,
                 spatialPanelExpanded,
-                crossLayerExpanded,
+                // if callers do not explicitly set crossLayerExpanded flag when making the loadFilter call,
+                // a bug will appear on the second re-render of the QueryPanel.CrossLayerFilter section
+                // specifically the crossLayer is not persisted between repeated accesses to the panel.
+                // Unburden callers to add the flag always, set it if they do and set from initial state as fallback.
+                // Do not let it become undefined.
+                crossLayerExpanded: (action?.filter?.crossLayerExpanded || cleanInitialState.crossLayerExpanded),
```
Comment (MV88) [inline web/client/reducers/queryform.js:603]: please add unit test

**Review by MV88 — CHANGES_REQUESTED** [review]
it works, please add a unit tests,

generally speaking all code that is used in 

- actions
- reducers
- selectors
- epics
- utils
- and sometimes components 
- plus other js files

should be covered by a  unit test

File: web/client/reducers/queryform.js (hunk 1 above)
Comment (Igi-ID) [reply web/client/reducers/queryform.js:603]: Added in https://github.com/geosolutions-it/MapStore2/pull/10022/commits/d60c6e0bb46e0d6fd33ed3ed646b909d9d855331

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo please test it in DEV

**Comment by ElenaGallo** [comment]
Test passed, @Igi-ID please backport to 2024.01.xx. Thanks

**Pull request merged**